x-pack/plugins/cross_cluster_replication/server/routes/api/follower_index.js

```diff
@@ -100,6 +107,37 @@ export const registerFollowerIndexRoutes = (server) => {
     },
   });
 
+  /**
+   * Edit a follower index
+   */
+  server.route({
+    path: `${API_BASE_PATH}/follower_indices/{id}`,
+    method: 'PUT',
+    config: {
+      pre: [ licensePreRouting ]
+    },
+    handler: async (request) => {
+      const callWithRequest = callWithRequestFactory(server, request);
+      const { id: _id } = request.params;
+      const body = removeEmptyFields(serializeAdvancedSettings(request.payload));
+
+      try {
+        /**
+         * We need to first pause the follower and then resume it passing the advanced settings
+         */
+        // Pause follower
+        await callWithRequest('ccr.pauseFollowerIndex', { id: _id });
```

Comment:
Would it make sense to wrap the `pause` and `resume` requests in individual try/catches, and then respond with errors that provides more insight into what happened? For example, if the pause fails, we could add a special code that will allow the UI to explain "Your follower index could not be paused, so it wasn't updated" and then provide the underlying ES error.

Reply:
I think this is a nice enhancement, but deferring since it's an edge case